src/transformers/pipelines/zero_shot_object_detection.py

```diff
@@ -0,0 +1,268 @@
+
+        requires_backends(self, "vision")
+        self.check_model_type(MODEL_FOR_ZERO_SHOT_OBJECT_DETECTION_MAPPING)
+
+    def __call__(
+        self,
+        images: Union[str, List[str], "Image.Image", List["Image.Image"]],
+        text_queries: Union[str, List[str], List[List[str]]] = None,
+        **kwargs
+    ):
+        """
+        Detect objects (bounding boxes & classes) in the image(s) passed as inputs.
+
+        Args:
+            images (`str`, `List[str]`, `PIL.Image` or `List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+            text_queries (`str` or `List[str]` or `List[List[str]]`): Text queries to query the target image with.
+            If given multiple images, `text_queries` should be provided as a list of lists, where each nested list
+            contains the text queries for the corresponding image.
+
+        Return:
+            A list of dictionaries containing prediction results, one dictionary per each input image. The dictionaries
+            contain the following keys:
+
+            - **labels** (`List[str]`) -- The list of text queries corresponding to the found objects.
+            - **scores** (`List[float]`) -- The list of scores corresponding to each object (between 0 and 1).
+            - **boxes** (`List[Dict[str,int]]`) -- A nested list of bounding boxes of all detected objects in image's
+              original size. Each list contains a dictionary with `x_min`, `x_max`, `y_min`, `y_max` keys.
+        """
+        if isinstance(text_queries, str) or (isinstance(text_queries, List) and not isinstance(text_queries[0], List)):
+            if isinstance(images, (str, Image.Image)):
+                inputs = {"images": images, "text_queries": text_queries}
+            elif isinstance(images, List):
+                assert len(images) == 1, "Input text_queries and images must have correspondance"
+                inputs = {"images": images[0], "text_queries": text_queries}
+            else:
+                raise TypeError(f"Innapropriate type of images: {type(images)}")
+
+        elif isinstance(text_queries, str) or (isinstance(text_queries, List) and isinstance(text_queries[0], List)):
+            if isinstance(images, (Image.Image, str)):
+                images = [images]
+            assert len(images) == len(text_queries), "Input text_queries and images must have correspondance"
+            inputs = {"images": images, "text_queries": text_queries}
+        else:
+            """
+            Supports the following format
+            - {"images": images, "text_queries": text_queries}
+            """
+            inputs = images
+        results = super().__call__(inputs, **kwargs)
+        return results
+
+    def _sanitize_parameters(self, **kwargs):
```

Comment:
The threshold param flows in via kwargs. You can also refer [zero_shot_image_classification pipeline](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/zero_shot_image_classification.py#:~:text=def%20_sanitize_parameters(,return%20preprocess_params%2C%20%7B%7D%2C%20%7B%7D)